src/main/java/io/xpring/xrpl/model/XRPMemo.java

```diff
@@ -0,0 +1,61 @@
+package io.xpring.xrpl.model;
+
+import com.google.protobuf.ByteString;
+import org.xrpl.rpc.v1.Memo;
+import org.immutables.value.Value;
+
+import javax.annotation.Nullable;
+
+/**
+ * Represents a memo on the XRPLedger.
+ * @see "https://xrpl.org/transaction-common-fields.html#memos-field"
+ */
+@Value.Immutable
+public interface XRPMemo {
+    static ImmutableXRPMemo.Builder builder() {
+        return ImmutableXRPMemo.builder();
+    }
+
+    /**
+     * @return Arbitrary hex value, conventionally containing the content of the memo.
+     */
+    @Nullable  byte[] data();
+
+    /**
+     * @return Hex value representing characters allowed in URLs.
+     * Conventionally containing information on how the memo is encoded, for example as a MIME type.
+     */
+    @Nullable byte[] format();
+
+    /**
+     * @return Hex value representing characters allowed in URLs.
+     * Conventionally, a unique relation (according to RFC 5988) that defines the format of this memo.
+     */
+    @Nullable byte[] type();
+
+    static XRPMemo from(Memo memo) {
+        byte[] data;
+        byte[] format;
+        byte[] type;
+        if (memo.getMemoData().getValue() == ByteString.EMPTY) {
```

Comment:
I'm surprised that `==` WAI. `==` in java should compare object addresses and I'm surprised an arbitrary `EMPTY` ByteString has the same object address as another `EMPTY` byte string. 

We should probably use `.equals()` here:P
```
if (memo.getMemoData.getValue().equals(ByteString.EMPTY) {
```

Reply:
I think that the reason this is WAI is that when you construct a default, empty `MemoData` object, its bytestring value is explicitly set to that constant `ByteString.EMPTY` in the constructor. `.equals` sounds good to me and is also easier to read in the ternary operator